src/display.rs

```diff
@@ -178,8 +180,10 @@ impl Display {
             height: viewport_size.height.0 as f32,
             cell_width: cell_width as f32,
             cell_height: cell_height as f32,
-            padding_x: config.padding().x as f32,
-            padding_y: config.padding().y as f32,
+            padding_top: top as f32,
+            padding_right: right as f32,
+            padding_bottom: bottom as f32,
+            padding_left: left as f32,
```

Comment:
At this point it would probably be a better idea to store `padding` on the Display rather than having four different fields. Accessing `self.padding.left` should not be any more complicated than `self.padding_left` either.

Reply:
This impacts multiple different structs like SizeInfo too, I would propose changing that for most of them if it makes parameters and struct definitions simpler (it should).